lib/owncloud/test_sharePropagation.py

```diff
@@ -0,0 +1,262 @@
++-------------+-------------------------+-------------------------+----------------------+
+| 11          | verify etag is the same | verify propagation      | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+| 12          |                         | R2 upload in shared dir |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 13          | verify propagation      | verify propagation      | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+| 14          |                         |                         | upload in shared dir |
++-------------+-------------------------+-------------------------+----------------------+
+| 15          | verify propagation      | verify propagation      | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+| 16          |                         | R2 unshares folder      |                      |
++-------------+-------------------------+-------------------------+----------------------+
+| 17          | verify etag is the same | verify etag is the same | verify propagation   |
++-------------+-------------------------+-------------------------+----------------------+
+
+Remove the sleep(x) once https://github.com/owncloud/client/issues/4160 has a resolution
+"""
+
+from smashbox.utilities import *
+import itertools
+import os.path
+import re
+import time
+
+def parse_worker_number(worker_name):
+    match = re.search(r'(\d+)$', worker_name)
+    if match is not None:
+        return int(match.group())
+    else:
+        return None
+
+@add_worker
+def setup(step):
+
+    step(1, 'create test users')
+    reset_owncloud_account(num_test_users=5)
+    check_users(5)
+
+    reset_rundir()
+    reset_server_log_file()
+
+    step(18, 'Validate server log file is clean')
+
+    d = make_workdir()
+    scrape_log_file(d)
+
+@add_worker
+def owner(step):
+
+    user = '%s%i' % (config.oc_account_name, 1)
+
+    step (2, 'Create workdir')
+    d = make_workdir()
+
+    mkdir(os.path.join(d, 'test'))
+    mkdir(os.path.join(d, 'test', 'sub'))
+    run_ocsync(d, user_num=1)
+
+    client = get_oc_api()
+    client.login(user, config.oc_account_password)
+    # make sure folder is shared
+    user2 = '%s%i' % (config.oc_account_name, 2)
+    share1_data = client.share_file_with_user('/test', user2, perms=31)
+    fatal_check(share1_data, 'failed sharing a file with %s' % (user2,))
+
+    user3 = '%s%i' % (config.oc_account_name, 3)
+    share2_data = client.share_file_with_user('/test', user3, perms=31)
+    fatal_check(share2_data, 'failed sharing a file with %s' % (user3,))
+
+    root_etag = client.file_info('/').get_etag()
+
+    step(3, 'Upload file')
+    createfile(os.path.join(d, 'test', 'test.txt'), '1', count=1000, bs=10)
+    time.sleep(3)
```

Comment:
https://github.com/owncloud/client/issues/4160 the issue has been fixed. So you can remove it when you use a git clone of the client or wait for the next daily build tomorrow